src/main/java/com/notably/model/block/BlockImpl.java

```diff
@@ -0,0 +1,123 @@
+package com.notably.model.block;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Represents a Block in the Notably data structure.
+ */
+public class BlockImpl implements Block {
+    private List<Block> children;
+    private Title title;
+    private Body body;
+    private Block parent;
+
+    /**
+     * Initializes a default block.
+     * Used when creating a root block and creating a block without the optional body argument.
+     */
+    public BlockImpl(Block parent, Title title) {
+        this.title = title;
+        this.parent = parent;
```

Comment:
Just FYI, you can call `this(parent, title, null)` too!

Reply:
Is the `null` Body intended, though? Perhaps you would want to create an "empty" `Body` instead?

Reply:
I did intend to use null for `body` and `children` but I see where you coming from and I think it makes a lot more sense. I'll update it